### PR TITLE
[TTAHUB-513] add percentage of recipients served to dashboard

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change

Add new tile to dashboard overview widget displaying the percentage of recipients served an activity report out of total unique recipients. 

Add new filters to the grant scopes so that certain AR filters would also change this number.

## How to test
 View the dashboard overview and check the numbers. Add applicable filters and verify any changed numbers.

## Issue(s)

* https://ocio-jira.acf.hhs.gov/browse/TTAHUB-513

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: src/scopes/grants/startDate.js [hunk 1]
```diff
@@ -1,26 +1,100 @@
...
+    if (!range.split) {
+      return acc;
+    }
+
+    const [sd, ed] = range.split('-');
+    if (!sd || !ed) {
+      return acc;
+    }
+
+    return [
```
Comment (thewatermethod) [inline src/scopes/grants/startDate.js:31]: I would not describe myself as incredibly confident that this is the best way to do this, it seems to pass my test cases but would love a second look at this

File: src/scopes/grants/grantNumber.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import { Op } from 'sequelize';
+
+export function withGrantNumber(grantNumber) {
+  const normalizedgrantNumber = `%${grantNumber}%`;
```
Comment (jasalisbury) [inline src/scopes/grants/grantNumber.js:4]: We probably need to use `sequelize.escape` here

File: src/scopes/grants/programSpecialist.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Op } from 'sequelize';
+
+export function withProgramSpecialist(name) {
+  const normalizedName = `%${name}%`;
```
Comment (jasalisbury) [inline src/scopes/grants/programSpecialist.js:4]: Another `sequelize.escape`

File: src/scopes/grants/startDate.js [hunk 4]
```diff
@@ -1,26 +1,100 @@
...
+export function afterStartDate(date) {
   return {
     [Op.and]: {
       [Op.or]: compareDate(date, 'startDate', Op.gt),
     },
   };
 }
 
-export function withinGrantStartDates(dates) {
+export function withinStartDates(dates) {
```
Comment (jasalisbury) [inline src/scopes/grants/startDate.js:20]: Should this be renamed to something like `activeWithinDates`? Something letting the user of the function know this scope looks for active grants between `dates`?

File: src/scopes/grants/startDate.js (hunk 1 above)
Comment (jasalisbury) [reply src/scopes/grants/startDate.js:31]: Can we just check if the grant start date is before the passed in end date and the grant end date is after the passed in end date? I think I have all cases in the following screenshot, a grant can start before, during or after the date range. Any grants that start after the date range shouldn't be included. If a grant starts during a date range it is always included. If a grant starts before the date range it has to end after the start of the date range.
Screenshot from 2022-01-26 11-38-03

File: src/scopes/grants/utils.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+  return searchTerms.map((term) => sequelize.literal(`"Grant"."id" ${operator} (${baseQuery} ${comparator} ${sequelize.escape(`%${term}%`)})`));
+}
+
+export function expandArrayContains(key, array, exclude) {
+  const comparator = exclude ? Op.notILike : Op.iLike;
+  const scopes = array.map((member) => {
+    const normalizedMember = `%${member}%`;
+    return {
+      [key]: {
+        [comparator]: normalizedMember, // sequelize escapes this automatically :)
```
Comment (jasalisbury) [inline src/scopes/grants/utils.js:15]: :tada:

**Review by jasalisbury — APPROVED** [review]

**Comment by thewatermethod** [comment]
Changed the percentage field based on discussion in Slack https://adhoc.slack.com/archives/C022R5301L4/p1643652368769839

**Review by GarrettEHill — APPROVED** [review]

**Comment by thewatermethod** [comment]
FYI, pending further review changes, this is waiting for tooltip text from patrice

**Comment by thewatermethod** [comment]
This has been demo'd and approved, just waiting for Wednesday so that the change can be communicated to the user base before merging.

**Pull request merged**